gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -743,6 +743,49 @@ protected GoogleCloudStorageItemInfo getInitialMetadata() {
     };
   }
 
+  public SeekableByteChannel open(
+          final GoogleCloudStorageItemInfo itemInfo, GoogleCloudStorageReadOptions readOptions)
+          throws IOException {
+    logger.atFiner().log("open(%s, %s)", itemInfo.getResourceId(), readOptions);
+    Preconditions.checkArgument(
+            itemInfo.getResourceId().isStorageObject(), "Expected full StorageObject id, got %s", itemInfo.getResourceId());
+
+    if (storageOptions.isGrpcEnabled()) {
+      return GoogleCloudStorageGrpcReadChannel
+              .open(storageStubProvider, storage, errorExtractor, itemInfo, readOptions);
+    }
+
+    // The underlying channel doesn't initially read data, which means that we won't see a
```

Comment:
If itemInfo is availabe, this comment is not valid

Reply:
Will modify the comment